### PR TITLE
reqOrgMembership calls need to be preceded by reqToken

### LINKED ISSUE #16192: Internal Server Error on endpoint ​/orgs​/{org}​/teams - List an organization's teams
- Gitea version (or commit ref): 1.14.2
- Git version: Not applicable
- Operating system: Official Gitea docker container
  
  
  
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes - https://gitea.com/api/v1/orgs/gitea/teams
  - [ ] No
- Log gist: https://gist.github.com/anders-larsson/730d73cdc825061a3b5bfe3904cfdadb

## Description

Using the List an organization's teams endpoint gives Internal Server Error and prints the error in the included Gist.

### PR DESCRIPTION
**Pull request by zeripath**
Fix #16192

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: routers/api/v1/api.go [hunk 1]
```diff
@@ -992,7 +992,7 @@ func Routes() *web.Route {
 				m.Combo("", reqToken()).Get(org.ListTeams).
 					Post(reqOrgOwnership(), bind(api.CreateTeamOption{}), org.CreateTeam)
 				m.Get("/search", org.SearchTeam)
```
Comment (6543) [inline routers/api/v1/api.go]:
```suggestion
				m.Get("", org.ListTeams)
				m.Post("", reqOrgOwnership(), bind(api.CreateTeamOption{}), org.CreateTeam)
				m.Get("/search", org.SearchTeam)
```

Comment (6543) [reply routers/api/v1/api.go]: rm already checked & make it more readable

Comment (noerw) [reply routers/api/v1/api.go]: `reqOrgOwnership()` !== `reqOrgMembership()`?

**Review by noerw — COMMENTED** [review]
test fail is same as in https://github.com/go-gitea/gitea/pull/15909#issuecomment-842652900

**Comment by zeripath** [comment]
Thanks @noerw that appears to have sorted it.

**Review by noerw — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Pull request merged**